### ISSUE TITLE
Analysis of 'Space Engineers' source code by PVS-Studio static code analyzer 

### ISSUE THREAD

**Issue by InkViziT0r** [KeenSoftwareHouse/SpaceEngineers#471]
Development team of proprietary PVS-Studio C/C++/C# static analyzer presents their report on the source code of 'Space Engineers' in the article, which contains the review of the most suspicious code fragments they discovered.

You can read article on the official site:
http://www.viva64.com/en/b/0376/

**Comment by harag-on-steam** [comment]
The linked article seems incomplete. Surely there's a list with all the issues somewhere? Also, this repository received a significant update to its codebase with 2d57a96b8bbb. So the results are probably inaccurate.

**Comment by InkViziT0r** [comment]
Yes, the article doesn't provide the full list of the errors found by the analyzer. Since the repository received a significant update, the best thing to do would probably be to download the analyzer and check the newer version yourself. 
We could generate a temporary key for your team.
Please, contact us:  http://www.viva64.com/en/about-feedback/

**Comment by AlonzoTG** [comment]
I wanna +1 this. Static analysis should be a legal requirement for releasing any software for money. =P
